src/main/java/uk/gov/pay/adminusers/app/config/SecondFactorAuthConfiguration.java

```diff
@@ -0,0 +1,27 @@
+package uk.gov.pay.adminusers.app.config;
+
+import javax.validation.constraints.NotNull;
+
+public class SecondFactorAuthConfiguration {
+    @NotNull
+    private int timeWindowInSeconds;
+    
+    @NotNull
+    private int validTimeWindows;
+    
+    public int getTimeWindowInSeconds() {
+        return timeWindowInSeconds;
+    }
+
+    public int getValidTimeWindows() {
+        return validTimeWindows;
+    }
+
+    public long getTimeWindowInMillis() {
+        return timeWindowInSeconds * 1000L;
+    }
+
+    public int getGoogleAuthenticatorWindowSize() {
+        return (validTimeWindows * 2) - 1;
```

Comment:
Took me a while to figure out what this is doing. I figured it’s something like this:

* We want to accept codes for the current window and 4 past windows (total 5)
* The way the library works is it checks the current window, _n_/2 windows in the past and _n_/2 windows in the future, where _n_ is the specified number of time windows
* Therefore, this conversion is necessary to make sure we tell the library to use enough windows to get the requisite 4 past windows and the current one (i.e. 9)

Is that correct?

If so, I don’t think it really gains us anything to have properties representing both the 5 and the 9, since the 5 figure is only ever used to calculate the nine. We should just have one property and set it to 9. Stick a comment explaining what the 9 means in practice if there’s concern the rationale may not be clear.